Fix: Compare mode fails with 404 when target tag not yet created (#322) (#324)
AbsaOSS/generate-release-notes: CONTRIBUTING.md

    * Ensure the Pull Request description clearly outlines your solution.
    * Link your PR to the relevant _Issue_.

## Compare Mode Contract

When contributing changes that touch compare mode (the `from-tag-name` path):
- Both `tag-name` and `from-tag-name` must exist as git tags before the compare API is called.
- The action validates this via `DataMiner._validate_compare_mode_tags()` and exits with code 1 if either tag is absent.

> - Any change to this validation logic must update `docs/features/compare_mode.md` and the tests in `tests/unit/release_notes_generator/data/test_miner.py`.

Author: Solved.

    * Ensure the Pull Request description clearly outlines your solution.
    * Link your PR to the relevant _Issue_.

## Compare Mode Contract

> When contributing changes that touch compare mode (the `from-tag-name` path):

Reviewer: ```suggestion
When contributing changes that touch compare mode (once `from-tag-name` input is provided):
```

Author: Section removed.

> - Both `tag-name` and `from-tag-name` must exist as git tags before the compare API is called.

Reviewer: This is the first time, there is mentioned this crucial info. Should be in the error message, when the run failes.

Author: addressed.

- Input-level check: `ActionInputs.validate_compare_mode_tag_names()` (called from `validate_inputs()`) exits with code 1 if either input string is empty.
- Tag-existence check: `DataMiner._handle_compare_mode()` calls `repo.get_git_ref("tags/<tag>")` for both tags and exits with code 1 naming the absent tag if either lookup fails.

> - Any change to this validation logic must update `docs/features/compare_mode.md` and the tests in `tests/unit/release_notes_generator/data/test_miner.py`.

Reviewer: Why is please in the CONTRIBUTING file described, how the tags are checked? Other topics in this file are named as Branch Naming or Proposing New Features. This described section looks here odd to me.

Author: You are right, it copy of other location.
